### DISCUSSION TITLE
Installing MOOSE with python 3.10 and Paraview 5.13

### DISCUSSION THREAD

**Discussion by josebastiase** [idaholab/moose#29022] [Q&A Modules: General] [answered]
Hi all,

I’m currently working on setting up an environment to run MOOSE alongside ParaView 5.13.1, the latest version available from the ParaView website, which is compatible with Python 3.10. However, I’m running into issues because MOOSE requires a newer version of Python, and I’m facing dependency conflicts during environment creation.
Environment Details

MOOSE Version: 2024.10.06
ParaView Version: 5.13.1
Python Version: 3.10 (for compatibility with ParaView)

Problem

When I try to create a conda environment that includes both MOOSE and ParaView, I encounter errors indicating that the Python versions required by each are conflicting. ParaView 5.13.1 works with Python 3.10, but MOOSE seems to require a newer version.

Attempts So Far

Specified Python 3.10 in the environment, with MOOSE and ParaView installed together.
Tried alternative versions of MOOSE (e.g., 2022.09.30) that might support Python 3.10.
Considered installing ParaView and MOOSE in separate environments but would ideally like to keep them together.

Question

Has anyone successfully installed MOOSE with ParaView 5.13.1, or does anyone have suggestions for managing these Python version conflicts? Any guidance would be much appreciated!

Thanks in advance!

**Comment by GiudGiud** [comment]
Hello

Moose and Paraview have their own requirements and if we followed them as well as ours it would make distributing moose harder. So unfortunately we won't have a solution for what you are trying to do. 

What we advise is actually to install paraview in a separate conda environment. That ways the dependencies of both software can be met.

You can also install paraview from the website binaries rather than with conda. 

Guillaume

**Reply by milljm** [reply, marked as answer]
It's not a Python version, but probably some dependency that requires Ptyhon 3.10 _and_ something else that conflicts.

A little unrelated: Our packages support Python 3.7 through 3.11. However, I am working on a PR that drops 3.7 (EOL), and adds 3.12.

What you can do, is drop the installation of the entire moose-*** package suite, and instead install only a subset of it:

```bash
conda create -n testing moose-libmesh=2024.10.17 moose-tools mpich paraview
```

The real conflict comes from `moose-peacock`. Which requires many of the same dependencies as Paraview (VTK specifically). I am betting it is some library here that is the true source of the conflict.

With the above install line, you're getting everything _except_ Peacock.


Edit: Peacock... it is such a pain. We used to have to instruct folks to create a separate env just for that. So this comes as no surprise.

**Reply by josebastiase** [reply]
Thank you all for your comments.

@milljm your are a wizard! I was trying to run a visualization tool that I wrote, and indeed they have vtk libs. That was creating conflict with Peacock. Thank you :)

**Reply by milljm** [reply]
oh, sorry... I forgot one:

```bash
conda create -n testing moose-libmesh=2024.10.17 moose-wasp moose-tools mpich paraview
                                                 ^^^^^^^^^^
```
Needed for running tests, etc.
